src/consensus/pool/parent_ready_tracker.rs

```diff
@@ -56,39 +56,27 @@ impl ParentReadyTracker {
                 break;
             }
         }
-
         newly_certified
     }
 
     /// Marks the given slot as skipped.
     ///
     /// Returns a list of any newly connected parents.
-    pub fn mark_skipped(&mut self, slot: Slot) -> Vec<(Slot, BlockId)> {
-        let state = self.slot_state(slot);
+    pub fn mark_skipped(&mut self, marked_slot: Slot) -> SmallVec<[(Slot, BlockId); 1]> {
+        let state = self.slot_state(marked_slot);
         if state.skip {
-            return Vec::new();
+            return SmallVec::new();
         }
         state.skip = true;
 
-        // get newly connected future windows
-        let mut future_windows = SmallVec::<[Slot; 1]>::new();
-        for slot in slot.future_slots() {
-            if slot.is_start_of_window() {
-                future_windows.push(slot);
-            }
-            if !self.slot_state(slot).skip {
-                break;
-            }
-        }
-
         // find possible parents for future windows
         let mut potential_parents = SmallVec::<[BlockId; 1]>::new();
-
-        for s in slot.slots_in_window().filter(|s| *s <= slot).rev() {
-            let state = self.slot_state(s);
-            if s < slot {
+        let window_slots = marked_slot.slots_in_window();
```

Comment:
github will not let me leave a sensible suggestion for this block.  Is the following code equivalent and cleaner?

```rust
        let mut potential_parents = SmallVec::<[BlockId; 1]>::from_slice(state.ready_block_ids());
        let window_slots = marked_slot.slots_in_window();
        for slot in window_slots.filter(|s| s < &marked_slot).rev() {
            let state = self.slot_state(slot);
            for nf in &state.notar_fallbacks {
                potential_parents.push((slot, *nf));
            }
            if !state.skip {
                break;
            }
            potential_parents.extend_from_slice(state.ready_block_ids());
        }
```